sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala

```diff
@@ -428,53 +432,85 @@ class SessionCatalog(
    */
   def dropFunction(name: FunctionIdentifier): Unit = {
     val db = name.database.getOrElse(currentDb)
+    val qualified = name.copy(database = Some(db)).unquotedString
+    if (functionRegistry.functionExists(qualified)) {
+      // If we have loaded this function into FunctionRegistry,
+      // also drop it from there.
+      functionRegistry.dropFunction(qualified)
```

Comment:
I see... it's because in `lookupFunction` we actually load a persistent function into the registry as a temporary function...